### ISSUE TITLE
Error when run PVAnet_test 

### ISSUE THREAD

**Issue by autumnqin** [CharlesShang/TFFRCNN#33]
python dv.py --model ~/PVAnet_iter_70000.ckpt --net PVAnet_test 

NotFoundError (see above for traceback): Key conv5_4/last_bn_scale/beta not found in checkpoint;

W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/alpha not found in checkpoint
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
	 [[Node: save/RestoreV2_473 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_473/tensor_names, save/RestoreV2_473/shape_and_slices)]]
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
	 [[Node: save/RestoreV2_473 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_473/tensor_names, save/RestoreV2_473/shape_and_slices)]]
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
	 [[Node: save/RestoreV2_473 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_473/tensor_names, save/RestoreV2_473/shape_and_slices)]]
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
	 [[Node: save/RestoreV2_473 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_473/tensor_names, save/RestoreV2_473/shape_and_slices)]]
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
	 [[Node: save/RestoreV2_473 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_473/tensor_names, save/RestoreV2_473/shape_and_slices)]]
W tensorflow/core/framework/op_kernel.cc:993] Not found: Key conv5_4/last_bn_scale/beta not found in checkpoint
	 [[Node: save/RestoreV2_473 = RestoreV2[dtypes=[DT_FLOAT], _device="/job:localhost/replica:0/task:0/cpu:0"](_recv_save/Const_0, save/RestoreV2_473/tensor_names, save/RestoreV2_473/shape_and_slices)]]

**Comment by jwnsu** [comment]
PVAnet is not ready yet. Test error is easy to fix. However, error rate is very high and not useable yet.

**Comment by CharlesShang** [comment]
Was it solved? 
There must be a typo in the test net. 
I'll run some debug in my spare time

**Comment by jwnsu** [comment]
test error is easy to resolve, however it has very high error (mAP is below 0.1) -- caffe version is one of most accurate models.  It seems conv2_1/1/conv layer is different from the paper, can also be caused by issue in pva_negation_block/v2.

**Comment by autumnqin** [comment]
train is different from test, and both are different from Caffe version

**Comment by baileyqbb** [comment]
Hi, there! I faced the same issue, but finally find the errors in PVAnet_test.py. Here is the lines I've changed to make the test work:
lines to comment: 
85 --> .scale(c_in=384, name='conv5_4/last_bn_scale')
118 --> (self.feed('convf_rpn', 'convf_2')
119 -->   .concat(axis=3, name='convf'))

lines to change:
92 --> (self.feed('downsample', 'conv4_4', 'upsamle')   --->  (self.feed('downsample', 'conv4_4')
121 --> (self.feed('convf', 'rois')  ---> (self.feed('convf_2', 'rois')
122 -->  .roi_pool(7, 7, 1.0/16, name='roi_pooling')  ---> .roi_pool(6, 6, 1.0/16, name='roi_pooling') 

Hope this will help!

BTW, as mentioned by @jwnsu, test error is quite high. I trained the model using VOC2012 data with 200000 iters, the mAP is about 0.03, which is totally unacceptable. I am gonna train the model with more iters, and meanwhile, check the PVAnet to see if it is the same with the caffe version mentioned by @autumnqin. 

Will update if any news.

**Comment by sevaxyz** [comment]
@baileyqbb Haven't you figured out the issue with the low mAP by any chance?

**Comment by yshen92** [comment]
Hi guys, can anyone please share some tips on improving the mAP?

**Comment by QueenJuliaZxx** [comment]
thank you very much @baileyqbb

**Comment by gentlebreeze1** [comment]
@baileyqbb @QueenJuliaZxx     do you run pva-faster-rcnn successfully?my pva.npy seem to be wrong.where do you download?what the script command do you run ?thanks for your help

**Comment by amirmgh1375** [comment]
@baileyqbb
thank you so much :+1: 
it works very well :100: :1st_place_medal:
